src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -72,11 +118,12 @@ interface ActiveWeb3ProviderProps {
 
 export function ActiveWeb3Provider({
   provider,
-  jsonRpcUrlMap,
+  jsonRpcUrlMap: propsJsonRpcUrlMap,
   defaultChainId: propsDefaultChainId,
   children,
 }: PropsWithChildren<ActiveWeb3ProviderProps>) {
   const onError = console.error
+  jsonRpcUrlMap = propsJsonRpcUrlMap
```

Comment:
i think this doesn't copy it, it just points a reference. you'd need to do a deep copy

Reply:
can we not just point a reference?

Reply:
if it's just a reference then why rename the prop?

Reply:
not sure if this is the right way to do it but i want to use jsonRpcUrlMap in a different file, so need to export it. 

creating a `export let jsonRpcUrlMap` & setting its value to the passed-in prop here
so need to disambiguate between the exported var and the prop

Reply:
yea wanna avoid this... shouldn't need to update jsonRpcUrlMap like that. if you do need to do it we should be using an atom